src/_pytest/recwarn.py

```diff
@@ -311,3 +340,5 @@ def found_str():
  Regex: {self.match_expr}
  Emitted warnings: {found_str()}"""
                         )
+                else:
+                    re_emit(self)
```

Comment:
Even if those methods are only used from inside `__exit__`, it seems very odd to me to define an instance method as a function inside here and pass `self` to it - why not make them normal methods of `WarningChecker` instead?

Reply:
Just used them to solve arg and mypy errors -- will look into mypy without the instance methods